File: test/lit/zig/tagged_union.zig
// RUN: %cot emit-cir %s | %FileCheck %s

// Features #053-054: Tagged union — Zig syntax
// Must be valid Zig — compilable by zig build-exe

const Shape = union(enum) {
    circle: i32,
    rect: i32,
    none,
};

pub fn makeCircle(r: i32) Shape {
    return Shape{ .circle = r };
}

// CHECK-LABEL: func.func @makeCircle
// CHECK: cir.union_init "circle"
// CHECK-SAME: !cir.tagged_union<"Shape"
// CHECK: return

File: test/lit/swift/tagged_union.swift
// RUN: %cot emit-cir %s | %FileCheck %s

// libsc: enums with associated values — Swift tagged unions
// enum Shape { case circle(Int32); case none } maps to !cir.tagged_union

enum Shape {
    case circle(Int32)
    case none
}

func makeCircle(_ r: Int32) -> Shape {
    return .circle(r)
}

// CHECK-LABEL: func.func @makeCircle
// CHECK-SAME: ({{%.*}}: i32) -> !cir.tagged_union<"Shape"
// CHECK: cir.union_init "circle"
// CHECK-SAME: i32
// CHECK: return
